.circleci/config.yml

```diff
@@ -229,6 +229,25 @@ jobs:
       - store_artifacts:
           path: /home/circleci/project/target/inspection-results
 
+  no-merge-commits:
+    docker:
+      - image: cimg/base:2022.10
+    steps:
+      - checkout
+      - run:
+          name: Check for merge commits
+          command: |
+            MERGE_COMMITS=$(git rev-list --merges master.."$CIRCLE_SHA1")
```

Comment:
Will this work if the top commit is not the merge?
Will this work if this is merged with an older master?

Reply:
1. Yes. The SHA will be of the last commit in the PR
2. I am not sure what you are asking

Reply:
1.
If the SHA will be of the last commit in the PR, how will this fail if the merge commit is not the last commit?
Example:
````
supplemental: fix violations (HEAD)
Merge with master (HEAD-1)
Issue: fix NPE (HEAD -2)
````

2.
`Will this work if this is merged with an older master?` => `Will this fail the CI if the PR is merged with an older master and not the latest?`

https://github.com/checkstyle/checkstyle/commits/master?after=e5b56f8dd158350df839198fd75d5eb40935011e+34&branch=master&qualified_name=refs%2Fheads%2Fmaster
Lets say 0d75b683a07c3863f2f463e78ef2e2ecad43cdd3 is master , but assume another commit snuck in while I was working and getting ready to push and server's master is really 379524f87603904f9f7272ea0e088d4b06821d1b .
Will this flag the merge even though I merged with 0d75b683a07c3863f2f463e78ef2e2ecad43cdd3  instead of 379524f87603904f9f7272ea0e088d4b06821d1b  ?

Reply:
>If the SHA will be of the last commit in the PR, how will this fail if the merge commit is not the last commit?

`start..finish` syntax checks all commits after `start` up to and including `finish`in the given range:
```
➜  checkstyle git:(issue-12319) git log -5
0bf87981c Pull #12345: make CI fail
059b96b69 Issue #12319: validate PR number when commit is for PR
cfb6899d4 Issue #12259: Create CI task to fail PR if merge commit exists
e5b56f8dd supplemental: ant switch in sevntu-checkstyle/sevntu.checkstyle#915
5768a5e30 minor: cleanups for ant-phase-verify
➜  checkstyle git:(issue-12319) git rev-list 5768a5e30..0bf87981c
0bf87981c766048276a61e9d1e5f9458f0781ec4
059b96b696970a4b0a2e36ed31e4294aa65657d2
cfb6899d4e22473a1458be096437a0fc443152a3
e5b56f8dd158350df839198fd75d5eb40935011e
```

Reply:
>Will this flag the merge even though I merged with https://github.com/checkstyle/checkstyle/commit/0d75b683a07c3863f2f463e78ef2e2ecad43cdd3 instead of https://github.com/checkstyle/checkstyle/commit/379524f87603904f9f7272ea0e088d4b06821d1b ?

Merge commit still exists in the PR, in the given commit range.